dbt/adapters/duckdb/environments/local.py

```diff
@@ -15,7 +16,12 @@ def __init__(self, cursor):
     def __getattr__(self, name):
         return getattr(self._cursor, name)
 
-    def execute(self, sql, bindings=None):
+    def execute(self, sql, bindings=None):   
+        #register all dfs 
+        #TODO is this okey to be here? 
```

Comment:
Yep I see what's going on here; will think of a place that we can move this block so that the registered DFs are available on the cursor via a call to the plugin

Reply:
So I think what we're going to need to do is to add a `configure_cursor` method to the `BasePlugin` interface and then call it in here: https://github.com/duckdb/dbt-duckdb/blob/master/dbt/adapters/duckdb/environments/__init__.py#L109

...that way the plugins can keep track of their own data frames and register them on each cursor as they get created (you can see from the `initialize_cursor` method that we have the same issue that the settings on each cursor do not persist from the parent connection to the child)

Reply:
So i made a bit refactoring as you said. I think it makes a lot of sense 

I had a problem that in the moment when i want to cretate my view i have to have different name of my df. So i moved creation here

Reply:
How do you do it normally, do you resolve such conversation or? 
By us normally is done by the person who initialized it? but otherwise i can do that too